robots/pkg/kubevirt/cmd/copy/jobs.go

```diff
@@ -0,0 +1,284 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package copy
+
+import (
+	"context"
+	"fmt"
+	"github.com/spf13/cobra"
+	"os"
+	"strings"
+
+	"github.com/google/go-github/github"
+	"k8s.io/test-infra/prow/config"
+	"sigs.k8s.io/yaml"
+
+	"kubevirt.io/project-infra/robots/pkg/kubevirt/cmd/flags"
+	github2 "kubevirt.io/project-infra/robots/pkg/kubevirt/github"
+	"kubevirt.io/project-infra/robots/pkg/kubevirt/jobconfig"
+	"kubevirt.io/project-infra/robots/pkg/kubevirt/log"
+	"kubevirt.io/project-infra/robots/pkg/querier"
+)
+
+type copyJobOptions struct {
+	jobConfigPathKubevirtPresubmits string
+	jobConfigPathKubevirtPeriodics  string
+}
+
+func (o copyJobOptions) Validate() error {
+	if _, err := os.Stat(o.jobConfigPathKubevirtPresubmits); os.IsNotExist(err) {
+		return fmt.Errorf("jobConfigPathKubevirtPresubmits is required: %v", err)
+	}
+	if _, err := os.Stat(o.jobConfigPathKubevirtPeriodics); os.IsNotExist(err) {
+		return fmt.Errorf("jobConfigPathKubevirtPeriodics is required: %v", err)
+	}
+	return nil
+}
+
+var copyJobsOpts = copyJobOptions{}
+
+const shortUse = "kubevirt copy jobs creates copies of the periodic and presubmit SIG jobs for latest kubevirtci providers"
+
+var copyJobsCommand = &cobra.Command{
+	Use:   "jobs",
+	Short: shortUse,
+	Long: fmt.Sprintf(`%s
+
+For each of the sigs (%s)
+it checks whether a job to run with the latest k8s version exists.
+If not, it copies the existing job for the previously latest kubevirtci provider and
+adjusts it to run with an eventually soon-to-be-existing new kubevirtci provider.
+
+Presubmit jobs will be created with
+
+	always_run: false
+	optional: true
+
+to avoid them failing all the time until the new provider is integrated into kubevirt/kubevirt.
+`, shortUse, strings.Join(jobconfig.SigNames, ", ")),
+	Run: run,
+}
+
+func CopyJobsCommand() *cobra.Command {
+	return copyJobsCommand
+}
+
+func init() {
+	copyJobsCommand.PersistentFlags().StringVar(&copyJobsOpts.jobConfigPathKubevirtPresubmits, "job-config-path-kubevirt-presubmits", "", "The path to the kubevirt presubmit job definitions")
+	copyJobsCommand.PersistentFlags().StringVar(&copyJobsOpts.jobConfigPathKubevirtPeriodics, "job-config-path-kubevirt-periodics", "", "The path to the kubevirt periodic job definitions")
+}
+
+func run(cmd *cobra.Command, args []string) {
+	flags.ParseFlagsOrExit(cmd, args, copyJobsOpts)
+
+	ctx := context.Background()
+	client := github2.NewGitHubClient(ctx)
+
+	releases, _, err := client.Repositories.ListReleases(ctx, "kubernetes", "kubernetes", nil)
+	if err != nil {
+		log.Log().Panicln(err)
+	}
+	releases = querier.ValidReleases(releases)
+	if len(releases) < 2 {
```

Comment:
Yes, no magic numbers, thanks for the catch!

Reply:
Thinking about what you said regarding cmd option: how would that help? I mean the number 2 is clearly defined by the requirement of a source and a target release needing to be present. Or do you mean we should not fetch the source and target release, but instead make them a command option so that we can copy jobs from any release to any release?

Reply:
If the latter is what you mean, we would need to fetch them from somewhere else? Where would that happen? In a parent command?

Reply:
You are totally right, in this case we need them to be always 2.